internal/cache/controller_obj_cache.go

```diff
@@ -2763,14 +2802,73 @@ func checkAndSetCloudType(client *clients.AviClient) bool {
 	return true
 }
 
+func checkIPAMForUsableNetworkLabels(client *clients.AviClient, ipamRefUri *string) bool {
+	// Donot check for labels in usable networks if a vipNetwork is provided by the user.
+	// In this case, the vipNetwork provided by the user will be used.
+	// 1. Prioritize user input vipetworkList, skip marker based selection if provided.
+	// 2. If not provided, check for markers in ipam's usable networks.
+	// 3. If marker based usable network is not available, keep vipNetworkList empty.
+	// 4. vipNetworkList can be empty only in WCP usecases, for all others, mark invalid configuration.
+
+	// 1. User input
+	if vipList, _ := lib.GetVipNetworkListEnv(); len(vipList) > 0 {
+		lib.SetVipNetworkList(vipList)
+		return true
+	}
+
+	// 2. Marker based, not valid for public clouds.
+	if ipamRefUri != nil {
```

Comment:
Same as above, is there a reason you want to expose this behavior outside of the `advancedL4` tag?